### ISSUE TITLE
libafs.ko module fails to load on AARCH64 with 'unsupported RELA relocation: 311'

### ISSUE THREAD

**Issue by semanticphilosopher** [NixOS/nixpkgs#284501]
### Describe the bug
The built openafs kernel module fails to load during startup on an AARCH64 processor. 
On investigation the following is present in `dmesg`
`module libafs: unsupported RELA relocation: 311`

### Steps To Reproduce
1. Using a recent release (eg `nixos-23.11.3496`) enable openafs, build and switch 
2. On reboot the `afsd.service` fails while evaluating the `afsd-pre-start` script with `insmod: ERROR: could not insert module /nix/store/...../openafs/libafs.ko.xz: Invalid module format`

### Expected behavior
That the `insmod` would successfully be loaded into the kernel and the `afsd.service` to run successfully.

### Additional context
Investigating  potential root causes lead to the possiblity that the binary has some  PIC instructions present.

Namely, using `readelf --relocs libafs.ko | egrep '(GOT|PLT|JU?MP_SLOT)'` which, apparently selects the PIC instructions was not - as expected - empty, but contained.

`000000077dc4  09e700000137 R_AARCH64_ADR_GOT 0000000000000000 key_type_keyring + 0`

This suggests that, somehow, during the building of the kernel module some PIC instructions became incorporated.

### Notify maintainers
@andersk @maggesi @spacefrogg

### Metadata
 ```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-linux"`
 - host os: `Linux 6.1.74, NixOS, 23.11 (Tapir)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by semanticphilosopher** [comment]
that was the only PIC reference in the module - I just didn't know how to track down where to search next

**Comment by cboylan** [comment]
As a heads up https://gerrit.openafs.org/#/c/15668/ should fix this problem.

**Comment by meffie** [comment]
OpenAFS 15668 is expected to be in included in OpenAFS 1.8.12 release.

**Comment by nessdoor** [comment]
The fix has been released on August 8th:
https://openafs.org/
https://openafs.org/dl/openafs/1.8.12/RELNOTES-1.8.12

**Comment by nessdoor** [comment]
I just noticed this issue had already been addressed in Nixpkgs by commit a6ccb54ac291b98dc00e96cef989c88fd38e3340, which introduced the upstream patch [15728](https://gerrit.openafs.org/15728) into our build process.

Closing this issue as solved.
